homeassistant/components/soma/__init__.py

```diff
@@ -123,3 +131,13 @@ def device_info(self):
             return
         self.current_position = 100 - response["position"]
         self.is_available = True
+        response = await self.hass.async_add_executor_job(
```

Comment:
If ` self.is_available = False` then we should skip this as we already know that we will not get the data.

Reply:
Why is that? I thought that flag was just for the UI to show the device as not reachable. My thinking was that I will mark the device as unavailable if communication fails for some reason. Then at a later time when communication is restored it would just come back as available. Since the end devices are connected over BLE packet loss at wireless range limits is sometimes enough to cause temporary loss of communications.

Reply:
When marked as unavailable, we won't store any state data, as the state is unavailable. So you won't need to fetch any other data either.

Reply:
If I have marked a device as unavailable is there a way to make it available again automatically. My thinking was that I would still periodically try to read the device state and if it succeeds then I can make the device available again.

Reply:
If a device is set to polling, Home Assistant will ask for an update based on an interval. Once you can get data again, set available=True.